### PR TITLE
Enable site isolation by default when deploying

### LINKED ISSUE #9: Support site isolation
We can use the site isolation parameter of the API. This is important for public repo's and ensures security by preventing access to other sites.

_Pseudocode:_ at https://github.com/ryangjchandler/forge-previewer/blob/main/app/Commands/DeployCommand.php#L177

```diff
-        $site = $this->forge->createSite($server->id, [
+        $data = [
             'domain' => $domain,
             'project_type' => 'php',
             'php_version' => $this->option('php-version'),
             'directory' => '/public'
-        ]);
+        ];
+
+        if (!$this->option('no-isolation')) {
+            $this->information('Enabling site isolation');
+
+            $data['isolation'] = true;
+            $data['username'] = str($this->getBranchName())->slug();
+        }
+
+        $site = $this->forge->createSite($server->id, $data);
 
         $this->information('Installing Git repository');
```

### PR DESCRIPTION
**Pull request by Illizian**
Closes #9

### CODE REVIEW

**Comment by Illizian** [comment]
Just need to check if we need to supplement this within the destroy command, as probably need to delete the user too :thinking: 

*Update:* There does not appear to be an option for managing this in either the Forge UI or the API.

**Comment by ryangjchandler** [comment]
Can we flip the logic round so that it's an opt-in instead of an opt-out?

**Comment by Illizian** [comment]
> Can we flip the logic round so that it's an opt-in instead of an opt-out?

This has been updated. It's now under the new flag `--site-isolation` :+1:

File: app/Commands/DeployCommand.php [hunk 1]
```diff
@@ -31,6 +31,7 @@ class DeployCommand extends Command
         {--command=* : A command you would like to execute on the site, e.g. php artisan db:seed.}
         {--edit-env=* : The colon-separated name and value that will be added/updated in the site\'s environment, e.g. "MY_API_KEY:my_api_key_value".}
         {--scheduler : Setup a cronjob to run Laravel\'s scheduler.}
+        {--site-isolation : Enable site isolation.}
```
Comment (ryangjchandler) [inline app/Commands/DeployCommand.php]:
```suggestion
        {--isolate : Enable site isolation.}
```

File: app/Commands/DeployCommand.php [hunk 2]
```diff
@@ -174,12 +175,21 @@ protected function findOrCreateSite(Server $server): Site
...
-        $site = $this->forge->createSite($server->id, [
+        $data = [
             'domain' => $domain,
             'project_type' => 'php',
             'php_version' => $this->option('php-version'),
             'directory' => '/public'
-        ]);
+        ];
+
+        if ($this->option('site-isolation')) {
```
Comment (ryangjchandler) [inline app/Commands/DeployCommand.php]:
```suggestion
        if ($this->option('isolate')) {
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

**Pull request merged**